### PR TITLE
feat: Remove `requirements.txt` for `pyproject.toml`

### PR DESCRIPTION
**Pull request by LeoFrancoCrespo**
This PR also does the following:
- Updates the `README.md` to reference the new environment setup document. 
- Updates the action file to use the toml file to install the dependencies instead of requirements.txt

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+version = "0.0.1"
+requires-python = ">=3.11, <3.12"
+
+dependencies = [
+  "python-dotenv == 1.0.0",
+  "spotipy == 2.23.0",
+  "flask == 2.2.5",
+  "pandas == 2.1.0",
+  "dash==2.13.0"
+]
```
Comment (mihaelaaleks) [inline pyproject.toml:12]: Was missing only one dependency when I ran it: 

>  "Flask-WTF==1.2.1"

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+[project.optional-dependencies]
+dev = [
+    "ipykernel == 6.25.2",
+    "pytest == 7.4.2"
+]
```
Comment (mihaelaaleks) [inline pyproject.toml:17]: Is this expected during pip install of [dev]:  
image

File: README.md [hunk 3]
```diff
@@ -7,9 +7,9 @@ Activate the environment:
```
Comment (mihaelaaleks) [inline README.md:7]: I've updated the readme format after this PR came in, I'll include this when resolving the merge conflict

File: .gitignore [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # VSCode
```
Comment (mihaelaaleks) [inline .gitignore:1]: These probably should also be added to the gitignore right?
image

**Review by mihaelaaleks — CHANGES_REQUESTED** [review]
Solid change! Everything runs perfectly fine, also with the notebooks. I like the separation of a development environment too.

File: pyproject.toml (hunk 1 above)
Comment (LeoFrancoCrespo) [reply pyproject.toml:12]: Added

File: pyproject.toml (hunk 2 above)
Comment (LeoFrancoCrespo) [reply pyproject.toml:17]: Yes it is expected

File: README.md (hunk 3 above)
Comment (LeoFrancoCrespo) [reply README.md]: i've merged and resolved the conflics ty

File: .gitignore (hunk 4 above)
Comment (LeoFrancoCrespo) [reply .gitignore:1]: They are added to the gitignore

When you run the pip install locally from pyproject toml they get created as part of that install :)

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+[project]
+name = "spotify_playist_curator"
+version = "0.0.1"
+requires-python = ">=3.11, <3.12"
+
+dependencies = [
```
Comment (mihaelaaleks) [inline pyproject.toml:6]: 546

**Pull request merged**